### DISCUSSION TITLE
Error

### DISCUSSION THREAD

**Discussion by Adris87** [spatie/laravel-permission#2720] [Q&A]
![Screenshot_2024-09-15-15-43-19-32_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/a07c65e1-44a0-4f2c-bb11-94c1d2666567)

**Comment by drbyte** [comment]
Looks like you are passing $role as a string (`'2'`) instead of as an integer (`2`), so the lookup is being performed on the role `name`, instead of the role `id`. 
If you pass your `id` as an `int` then it will do the lookup on the role record ID, instead of trying to match on name.
